torch/utils/cpp_extension.py

```diff
@@ -2,18 +2,36 @@
 import os
 import re
 import subprocess
+import setuptools
 import sys
 import sysconfig
 import tempfile
 import warnings
 
+import torch
+
 from setuptools.command.build_ext import build_ext
 
+
+def _find_cuda_home():
```

Comment:
This logic is duplicated with `tools/setup_helpers/cuda.py`, though I am not sure there is a reasonable way to arrange for the same Python module to be made available in both `setup.py` and `torch/utils`. Not for this PR, but maybe this is a sign we really need a pure Python package which our build system and torch depend upon (this would solve other code duplication with ATen and PyTorch.)

Reply:
Yes I agree. I noted this in the PR summary btw.

Reply:
I think one behavior I don't see here which is in the `setup_helpers` version, is a probe in the default directory locations:

```
LINUX_HOME = '/usr/local/cuda'
WINDOWS_HOME = glob.glob('C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v*.*')
```

This behavior is very helpful and a lot of users rely on it.

Reply:
Ok I'll add this, thanks.